components/engine/integration-cli/daemon/daemon_swarm.go

```diff
@@ -198,6 +198,23 @@ func (d *Swarm) CheckServiceTasksInState(service string, state swarm.TaskState,
 	}
 }
 
+// CheckServiceTasksInStateWithError returns the number of tasks with a matching state,
+// and optional message substring.
+func (d *Swarm) CheckServiceTasksInStateWithError(service string, state swarm.TaskState, errorMessage string) func(*check.C) (interface{}, check.CommentInterface) {
+	return func(c *check.C) (interface{}, check.CommentInterface) {
+		tasks := d.GetServiceTasks(c, service)
+		var count int
+		for _, task := range tasks {
+			if task.Status.State == state {
+				if errorMessage == "" || strings.Contains(task.Status.Err, errorMessage) {
```

Comment:
@nishanttotla is this a API change that we have to worry about?

Reply:
This is an API change for sure, but I don't think we have to worry too much about it. In case someone outside is consuming this field, their workflow will break but it's not a huge change and not hard to fix.

@andrewhsu let me know if we haven't updated this in the release notes yet, I can help do that.

Reply:
Information is now output in another field, so yes, slight API change in that respect (also why I originally left it out in my summary of changes to cherry-pick), but probably ok if we mention it